File: app/src/main/java/ru/org/adons/cblock/ui/view/IMainListener.java
package ru.org.adons.cblock.ui.view;

import ru.org.adons.cblock.ui.base.IBaseFragmentListener;

/**
 * Listen {@link MainFragment} and notify {@link MainActivity}
 */
interface IMainListener extends IBaseFragmentListener {

    void showAddFragment();

}

File: app/src/main/java/mobile/addons/cblock/ui/activity/IMainListener.kt
package mobile.addons.cblock.ui.activity

import mobile.addons.cblock.ui.view.add.AddFragment
import mobile.addons.cblock.ui.view.main.MainFragment

/**
 * Listen [MainFragment], [AddFragment] and notify [MainActivity]
 */
interface IMainListener {

    val mainComponent: MainComponent

    fun showAddFragment()

    fun showProgress()

    fun hideProgress()

    fun showError(message: String)

    fun onBackPressed()

}
